yahtzee/sam/code/yahtzee.js

```diff
@@ -0,0 +1,44 @@
+import R from 'ramda'
+
+const scoreUpper = scoreVal => R.compose(
+  R.sum, 
+  R.filter(R.equals(scoreVal))
+)
+
+export const scoreOnes = scoreUpper(1)
+export const scoreTwos = scoreUpper(2)
+
+const diceCounts = R.countBy(R.identity)
+const diceCountsWith = predicate => R.compose(
+  R.filter(predicate), 
+  diceCounts
+)
+
+const diceWithAtleast = count => R.compose(Object.keys, diceCountsWith(c => c >= count))
+ 
+const totalKinds = reducer => ofAKind => R.compose(
```

Comment:
Not sure I like the "kinds" verbiage, but I derived it from "two of a kind, three of a kind..." domain speak.